src/Stripe.net/Infrastructure/Middleware/RequestStringBuilder.cs

```diff
@@ -1,61 +1,416 @@
+        /// <returns>The list of parameters</returns>
+        private static List<Parameter> FlattenParams(INestedOptions options)
+        {
+            return FlattenParamsOptions(options, null);
+        }
+
+        /// <summary>
+        /// Returns a list of parameters for a given options class. If a key prefix is provided, the
+        /// keys for the new parameters will be nested under the key prefix. E.g. if the key prefix
+        /// `foo` is passed and the options class contains a parameter `bar`, then a parameter
+        /// with key `foo[bar]` will be returned.
+        /// </summary>
+        /// <param name="options">The options class for which to create the list of parameters.</param>
+        /// <param name="keyPrefix">The key under which new keys should be nested, if any.</param>
+        /// <returns>The list of parameters</returns>
+        private static List<Parameter> FlattenParamsOptions(INestedOptions options, string keyPrefix)
+        {
+            List<Parameter> flatParams = new List<Parameter>();
+            if (options == null)
+            {
+                return flatParams;
+            }
+
+            foreach (var property in options.GetType().GetRuntimeProperties())
+            {
+                var value = property.GetValue(options, null);
+                if (value == null)
+                {
+                    continue;
+                }
+
+                var attribute = property.GetCustomAttribute<JsonPropertyAttribute>();
+                if (attribute == null)
+                {
+                    continue;
+                }
+
+                string key = attribute.PropertyName;
+                string newPrefix = NewPrefix(key, keyPrefix);
+
+                flatParams.AddRange(FlattenParamsValue(value, newPrefix));
+            }
+
+            return flatParams;
+        }
+
+        /// <summary>
+        /// Returns a list of parameters for a given value. The value can be basically anything, as
+        /// long as it can be encoded in some way.
+        /// </summary>
+        /// <param name="value">The value for which to create the list of parameters.</param>
+        /// <param name="keyPrefix">The key under which new keys should be nested, if any.</param>
+        /// <returns>The list of parameters</returns>
+        private static List<Parameter> FlattenParamsValue(object value, string keyPrefix)
+        {
+            List<Parameter> flatParams = null;
+
+            if (value is INestedOptions)
+            {
+                flatParams = FlattenParamsOptions((INestedOptions)value, keyPrefix);
+            }
+            else if (value is StripeDateFilter)
+            {
+                flatParams = FlattenParamsDateFilter((StripeDateFilter)value, keyPrefix);
+            }
+            else if (IsDictionary(value))
+            {
+                // Cast to Dictionary<string, object>
```

Comment:
Basically, this code is needed to satisfy the type system. If you have e.g. a `Dictionary{string, string}` and you try to pass it directly to a method with signature `Dictionary{string, object}`, it will fail because the compiler is unable to directly cast the dictionary (even though strings are objects). The reason is that it makes the code unsafe -- once you have a `Dictionary{string, object}`, you could e.g. add a new value that is not a string (cf. [this SO answer](https://stackoverflow.com/a/5886870/5307473)).

There may be a more elegant way to handle this, but with my limited knowledge of C#'s type system and reflection, this is the best I could come up with.